### ISSUE TITLE
RuntimeError: no running event loop sys

### ISSUE THREAD

**Issue by GreedyDragon81** [SpielmannSpiel/Godot-Private-Asset-Library#1]
I am getting this error,
{ RuntimeError: no running event loop
sys:1: RuntimeWarning: coroutine 'ProjectManager.load_projects' was never awaited.}

 I set it up from the docker_compose.yml. container is running but cant accsess the url. Url is using the url is using the private network address my other containers use. 8080 not in use. So im not sure what is going on.

**Comment by bison--** [comment]
Sorry for the late response, I am not used for people to actually comment on any of my github projects and haven't seen your issue, will look into it!

**Comment by bison--** [comment]
I need a bit more information
* which python version are you using?
* which operating system?
* which of the available methods of running it do you use?

**Comment by GreedyDragon81** [comment]
No worries! Thanks for responding. I am still hoping to get this up and running. I will reinstall tomorrow or Sunday and provide you with the requested information

**Comment by GreedyDragon81** [comment]
Sorry for the late response, but here's everything it shows when I try the docker install.

Unbuntu 22.04 - server
Latest Python : python3 3.12.3-0ubuntu2


I tried docker compose up of 

`docker compose -f docker-compose-local.yml `

up and also 

`docker compose -f docker-compose-local.yml build godot_asset_library
`


':~/git/Godot-Private-Asset-Library$ docker compose -f Docker-compose-local.yml up'
>WARN[0000] /home/dread81/git/Godot-Private-Asset-Library/Docker-compose-local.yml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion 
Compose can now delegate builds to bake for better performance.
 To do so, set COMPOSE_BAKE=true.
[+] Building 58.8s (15/15) FINISHED                              docker:default
 => [godot_private_asset_library internal] load build definition from Doc  0.1s
 => => transferring dockerfile: 776B                                       0.0s
 => [godot_private_asset_library] resolve image config for docker-image:/  1.5s
 => [godot_private_asset_library] docker-image://docker.io/docker/dockerf  4.4s
 => => resolve docker.io/docker/dockerfile:1@sha256:4c68376a702446fc3c79a  0.1s
 => => sha256:dfbaeac3c8884ccc4819c898cb14b39a17a21f4d13e 1.26kB / 1.26kB  0.0s
 => => sha256:07501dea36dacac10d4bcb55e0c725daaea8ec0b0 13.14MB / 13.14MB  4.0s
 => => sha256:4c68376a702446fc3c79af22de146a148bc3367e73c 8.40kB / 8.40kB  0.0s
 => => sha256:321a8dfcd7d7961eba146dbd133fae9c59d16eeaebc73b8 850B / 850B  0.0s
 => => extracting sha256:07501dea36dacac10d4bcb55e0c725daaea8ec0b07e17c70  0.1s
 => [godot_private_asset_library internal] load metadata for docker.io/li  1.3s
 => [godot_private_asset_library internal] load .dockerignore              0.1s
 => => transferring context: 314B                                          0.0s
 => [godot_private_asset_library internal] load build context              0.1s
 => => transferring context: 1.96MB                                        0.0s
 => [godot_private_asset_library 1/7] FROM docker.io/library/python:3.10  12.2s
 => => resolve docker.io/library/python:3.10-slim@sha256:06f6d69d229bb55f  0.1s
 => => sha256:a8053d65de8edac0d12f24eb1817ab2bec12d899c99 3.51MB / 3.51MB  1.9s
 => => sha256:06f6d69d229bb55fab83dded514e54eede977e33e92 9.13kB / 9.13kB  0.0s
 => => sha256:3fb3111ff5ff4efaf8091eb42d6626e2f0940ad36fb 1.75kB / 1.75kB  0.0s
 => => sha256:643092d2b2968a7c0d7d64fbc517f9222cd399fb332 5.29kB / 5.29kB  0.0s
 => => sha256:6e909acdb790c5a1989d9cfc795fda5a246ad6664 28.20MB / 28.20MB  9.9s
 => => sha256:806331b0d26083206f286a87d0bc51fd32227a07c 15.64MB / 15.64MB  7.7s
 => => sha256:ce054015c4fbe663550dd11a22748c34152e91ebacb5b32 250B / 250B  2.2s
 => => extracting sha256:6e909acdb790c5a1989d9cfc795fda5a246ad6664bb27b5c  0.8s
 => => extracting sha256:a8053d65de8edac0d12f24eb1817ab2bec12d899c992a2f9  0.1s
 => => extracting sha256:806331b0d26083206f286a87d0bc51fd32227a07c4747cb6  0.5s
 => => extracting sha256:ce054015c4fbe663550dd11a22748c34152e91ebacb5b327  0.0s
 => [godot_private_asset_library 2/7] WORKDIR /app                         0.3s
 => [godot_private_asset_library 3/7] RUN apt-get update -y      && apt-  18.4s
 => [godot_private_asset_library 4/7] COPY ./requirements.txt /requiremen  0.2s
 => [godot_private_asset_library 5/7] RUN pip install -U pip               4.9s
 => [godot_private_asset_library 6/7] RUN pip install --no-cache-dir --u  13.6s
 => [godot_private_asset_library 7/7] COPY . /app                          0.2s
 => [godot_private_asset_library] exporting to image                       0.8s
 => => exporting layers                                                    0.7s
 => => writing image sha256:70d87f8d4cf26cb0aef418a1050c0648049282fd72430  0.0s
 => => naming to docker.io/library/godot-private-asset-library-godot_priv  0.0s
 => [godot_private_asset_library] resolving provenance for metadata file   0.0s
[+] Running 3/3
 ✔ godot_private_asset_library                  Built                      0.0s 
 ✔ Network godot-private-asset-library_default  Created                    0.1s 
 ✔ Container godot_private_asset_library        Created                    0.2s 
Attaching to godot_private_asset_library
godot_private_asset_library  | 
godot_private_asset_library  |    FastAPI   Starting production server 🚀
godot_private_asset_library  |  
godot_private_asset_library  |              Searching for package file structure from directories with         
godot_private_asset_library  |              __init__.py files                                                  
godot_private_asset_library  | 
godot_private_asset_library  | ╭───────────────────── Traceback (most recent call last) ──────────────────────╮
godot_private_asset_library  | │ /usr/local/lib/python3.10/site-packages/fastapi_cli/cli.py:334 in run        │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │   331 │                                                                      │
godot_private_asset_library  | │   332 │   Otherwise, it uses the first [bold]FastAPI[/bold] app found in the │
godot_private_asset_library  | │   333 │   """                                                                │
godot_private_asset_library  | │ ❱ 334 │   _run(                                                              │
godot_private_asset_library  | │   335 │   │   path=path,                                                     │
godot_private_asset_library  | │   336 │   │   host=host,                                                     │
godot_private_asset_library  | │   337 │   │   port=port,                                                     │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │ ╭─────────────── locals ───────────────╮                                     │
godot_private_asset_library  | │ │           app = None                 │                                     │
godot_private_asset_library  | │ │          host = '0.0.0.0'            │                                     │
godot_private_asset_library  | │ │          path = PosixPath('main.py') │                                     │
godot_private_asset_library  | │ │          port = 8080                 │                                     │
godot_private_asset_library  | │ │ proxy_headers = True                 │                                     │
godot_private_asset_library  | │ │        reload = False                │                                     │
godot_private_asset_library  | │ │     root_path = ''                   │                                     │
godot_private_asset_library  | │ │       workers = None                 │                                     │
godot_private_asset_library  | │ ╰──────────────────────────────────────╯                                     │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │ /usr/local/lib/python3.10/site-packages/fastapi_cli/cli.py:101 in _run       │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │    98 │   │   )                                                              │
godot_private_asset_library  | │    99 │   │                                                                  │
godot_private_asset_library  | │   100 │   │   try:                                                           │
godot_private_asset_library  | │ ❱ 101 │   │   │   import_data = get_import_data(path=path, app_name=app)     │
godot_private_asset_library  | │   102 │   │   except FastAPICLIException as e:                               │
godot_private_asset_library  | │   103 │   │   │   toolkit.print_line()                                       │
godot_private_asset_library  | │   104 │   │   │   toolkit.print(f"[error]{e}")                               │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │ ╭───────────────────────────────── locals ─────────────────────────────────╮ │
godot_private_asset_library  | │ │           app = None                                                     │ │
godot_private_asset_library  | │ │       command = 'run'                                                    │ │
godot_private_asset_library  | │ │          host = '0.0.0.0'                                                │ │
godot_private_asset_library  | │ │          path = PosixPath('main.py')                                     │ │
godot_private_asset_library  | │ │          port = 8080                                                     │ │
godot_private_asset_library  | │ │ proxy_headers = True                                                     │ │
godot_private_asset_library  | │ │        reload = False                                                    │ │
godot_private_asset_library  | │ │     root_path = ''                                                       │ │
godot_private_asset_library  | │ │   server_type = 'production'                                             │ │
godot_private_asset_library  | │ │       toolkit = <rich_toolkit.toolkit.RichToolkit object at              │ │
godot_private_asset_library  | │ │                 0x76a3d1125e70>                                          │ │
godot_private_asset_library  | │ │       workers = None                                                     │ │
godot_private_asset_library  | │ ╰──────────────────────────────────────────────────────────────────────────╯ │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │ /usr/local/lib/python3.10/site-packages/fastapi_cli/discover.py:126 in       │
godot_private_asset_library  | │ get_import_data                                                              │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │   123 │   │   raise FastAPICLIException(f"Path does not exist {path}")       │
godot_private_asset_library  | │   124 │   mod_data = get_module_data_from_path(path)                         │
godot_private_asset_library  | │   125 │   sys.path.insert(0, str(mod_data.extra_sys_path))                   │
godot_private_asset_library  | │ ❱ 126 │   use_app_name = get_app_name(mod_data=mod_data, app_name=app_name)  │
godot_private_asset_library  | │   127 │                                                                      │
godot_private_asset_library  | │   128 │   import_string = f"{mod_data.module_import_str}:{use_app_name}"     │
godot_private_asset_library  | │   129                                                                        │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │ ╭──────────────────────── locals ─────────────────────────╮                  │
godot_private_asset_library  | │ │ app_name = None                                         │                  │
godot_private_asset_library  | │ │ mod_data = ModuleData(                                  │                  │
godot_private_asset_library  | │ │            │   module_import_str='main',                │                  │
godot_private_asset_library  | │ │            │   extra_sys_path=PosixPath('/app'),        │                  │
godot_private_asset_library  | │ │            │   module_paths=[                           │                  │
godot_private_asset_library  | │ │            │   │   PosixPath('/app/main.py')            │                  │
godot_private_asset_library  | │ │            │   ]                                        │                  │
godot_private_asset_library  | │ │            )                                            │                  │
godot_private_asset_library  | │ │     path = PosixPath('main.py')                         │                  │
godot_private_asset_library  | │ ╰─────────────────────────────────────────────────────────╯                  │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │ /usr/local/lib/python3.10/site-packages/fastapi_cli/discover.py:70 in        │
godot_private_asset_library  | │ get_app_name                                                                 │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │    67                                                                        │
godot_private_asset_library  | │    68 def get_app_name(*, mod_data: ModuleData, app_name: Union[str, None] = │
godot_private_asset_library  | │    69 │   try:                                                               │
godot_private_asset_library  | │ ❱  70 │   │   mod = importlib.import_module(mod_data.module_import_str)      │
godot_private_asset_library  | │    71 │   except (ImportError, ValueError) as e:                             │
godot_private_asset_library  | │    72 │   │   logger.error(f"Import error: {e}")                             │
godot_private_asset_library  | │    73 │   │   logger.warning(                                                │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │ ╭──────────────────────── locals ─────────────────────────╮                  │
godot_private_asset_library  | │ │ app_name = None                                         │                  │
godot_private_asset_library  | │ │ mod_data = ModuleData(                                  │                  │
godot_private_asset_library  | │ │            │   module_import_str='main',                │                  │
godot_private_asset_library  | │ │            │   extra_sys_path=PosixPath('/app'),        │                  │
godot_private_asset_library  | │ │            │   module_paths=[                           │                  │
godot_private_asset_library  | │ │            │   │   PosixPath('/app/main.py')            │                  │
godot_private_asset_library  | │ │            │   ]                                        │                  │
godot_private_asset_library  | │ │            )                                            │                  │
godot_private_asset_library  | │ ╰─────────────────────────────────────────────────────────╯                  │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │ /usr/local/lib/python3.10/importlib/__init__.py:126 in import_module         │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │   123 │   │   │   if character != '.':                                       │
godot_private_asset_library  | │   124 │   │   │   │   break                                                  │
godot_private_asset_library  | │   125 │   │   │   level += 1                                                 │
godot_private_asset_library  | │ ❱ 126 │   return _bootstrap._gcd_import(name[level:], package, level)        │
godot_private_asset_library  | │   127                                                                        │
godot_private_asset_library  | │   128                                                                        │
godot_private_asset_library  | │   129 _RELOADING = {}                                                        │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │ ╭───── locals ─────╮                                                         │
godot_private_asset_library  | │ │   level = 0      │                                                         │
godot_private_asset_library  | │ │    name = 'main' │                                                         │
godot_private_asset_library  | │ │ package = None   │                                                         │
godot_private_asset_library  | │ ╰──────────────────╯                                                         │
godot_private_asset_library  | │ in _gcd_import:1050                                                          │
godot_private_asset_library  | │ ╭───── locals ─────╮                                                         │
godot_private_asset_library  | │ │   level = 0      │                                                         │
godot_private_asset_library  | │ │    name = 'main' │                                                         │
godot_private_asset_library  | │ │ package = None   │                                                         │
godot_private_asset_library  | │ ╰──────────────────╯                                                         │
godot_private_asset_library  | │ in _find_and_load:1027                                                       │
godot_private_asset_library  | │ ╭────────────────── locals ──────────────────╮                               │
godot_private_asset_library  | │ │ module = <object object at 0x76a3d3404060> │                               │
godot_private_asset_library  | │ │   name = 'main'                            │                               │
godot_private_asset_library  | │ ╰────────────────────────────────────────────╯                               │
godot_private_asset_library  | │ in _find_and_load_unlocked:1006                                              │
godot_private_asset_library  | │ ╭───────────────────────────────── locals ─────────────────────────────────╮ │
godot_private_asset_library  | │ │   name = 'main'                                                          │ │
godot_private_asset_library  | │ │ parent = ''                                                              │ │
godot_private_asset_library  | │ │   path = None                                                            │ │
godot_private_asset_library  | │ │   spec = ModuleSpec(name='main',                                         │ │
godot_private_asset_library  | │ │          loader=<_frozen_importlib_external.SourceFileLoader object at   │ │
godot_private_asset_library  | │ │          0x76a3d1124df0>, origin='/app/main.py')                         │ │
godot_private_asset_library  | │ ╰──────────────────────────────────────────────────────────────────────────╯ │
godot_private_asset_library  | │ in _load_unlocked:688                                                        │
godot_private_asset_library  | │ ╭───────────────────────────────── locals ─────────────────────────────────╮ │
godot_private_asset_library  | │ │ module = <module 'main' from '/app/main.py'>                             │ │
godot_private_asset_library  | │ │   spec = ModuleSpec(name='main',                                         │ │
godot_private_asset_library  | │ │          loader=<_frozen_importlib_external.SourceFileLoader object at   │ │
godot_private_asset_library  | │ │          0x76a3d1124df0>, origin='/app/main.py')                         │ │
godot_private_asset_library  | │ ╰──────────────────────────────────────────────────────────────────────────╯ │
godot_private_asset_library  | │ in exec_module:883                                                           │
godot_private_asset_library  | │ ╭───────────────────────────────── locals ─────────────────────────────────╮ │
godot_private_asset_library  | │ │   code = <code object <module> at 0x76a3d11c85b0, file "/app/main.py",   │ │
godot_private_asset_library  | │ │          line 1>                                                         │ │
godot_private_asset_library  | │ │ module = <module 'main' from '/app/main.py'>                             │ │
godot_private_asset_library  | │ │   self = <_frozen_importlib_external.SourceFileLoader object at          │ │
godot_private_asset_library  | │ │          0x76a3d1124df0>                                                 │ │
godot_private_asset_library  | │ ╰──────────────────────────────────────────────────────────────────────────╯ │
godot_private_asset_library  | │ in _call_with_frames_removed:241                                             │
godot_private_asset_library  | │ ╭───────────────────────────────── locals ─────────────────────────────────╮ │
godot_private_asset_library  | │ │ args = (                                                                 │ │
godot_private_asset_library  | │ │        │   <code object <module> at 0x76a3d11c85b0, file "/app/main.py", │ │
godot_private_asset_library  | │ │        line 1>,                                                          │ │
godot_private_asset_library  | │ │        │   {                                                             │ │
godot_private_asset_library  | │ │        │   │   '__name__': 'main',                                       │ │
godot_private_asset_library  | │ │        │   │   '__doc__': None,                                          │ │
godot_private_asset_library  | │ │        │   │   '__package__': '',                                        │ │
godot_private_asset_library  | │ │        │   │   '__loader__':                                             │ │
godot_private_asset_library  | │ │        <_frozen_importlib_external.SourceFileLoader object at            │ │
godot_private_asset_library  | │ │        0x76a3d1124df0>,                                                  │ │
godot_private_asset_library  | │ │        │   │   '__spec__': ModuleSpec(name='main',                       │ │
godot_private_asset_library  | │ │        loader=<_frozen_importlib_external.SourceFileLoader object at     │ │
godot_private_asset_library  | │ │        0x76a3d1124df0>, origin='/app/main.py'),                          │ │
godot_private_asset_library  | │ │        │   │   '__file__': '/app/main.py',                               │ │
godot_private_asset_library  | │ │        │   │   '__cached__': '/app/__pycache__/main.cpython-310.pyc',    │ │
godot_private_asset_library  | │ │        │   │   '__builtins__': {                                         │ │
godot_private_asset_library  | │ │        │   │   │   '__name__': 'builtins',                               │ │
godot_private_asset_library  | │ │        │   │   │   '__doc__': 'Built-in functions, exceptions, and other │ │
godot_private_asset_library  | │ │        objects.\n\nNoteworthy: None is the `nil'+46,                     │ │
godot_private_asset_library  | │ │        │   │   │   '__package__': '',                                    │ │
godot_private_asset_library  | │ │        │   │   │   '__loader__': <class                                  │ │
godot_private_asset_library  | │ │        '_frozen_importlib.BuiltinImporter'>,                             │ │
godot_private_asset_library  | │ │        │   │   │   '__spec__': ModuleSpec(name='builtins', loader=<class │ │
godot_private_asset_library  | │ │        '_frozen_importlib.BuiltinImporter'>, origin='built-in'),         │ │
godot_private_asset_library  | │ │        │   │   │   '__build_class__': <built-in function                 │ │
godot_private_asset_library  | │ │        __build_class__>,                                                 │ │
godot_private_asset_library  | │ │        │   │   │   '__import__': <built-in function __import__>,         │ │
godot_private_asset_library  | │ │        │   │   │   'abs': <built-in function abs>,                       │ │
godot_private_asset_library  | │ │        │   │   │   'all': <built-in function all>,                       │ │
godot_private_asset_library  | │ │        │   │   │   'any': <built-in function any>,                       │ │
godot_private_asset_library  | │ │        │   │   │   ... +145                                              │ │
godot_private_asset_library  | │ │        │   │   },                                                        │ │
godot_private_asset_library  | │ │        │   │   'os': <module 'os' from                                   │ │
godot_private_asset_library  | │ │        '/usr/local/lib/python3.10/os.py'>,                               │ │
godot_private_asset_library  | │ │        │   │   'time': <module 'time' (built-in)>,                       │ │
godot_private_asset_library  | │ │        │   │   ... +12                                                   │ │
godot_private_asset_library  | │ │        │   }                                                             │ │
godot_private_asset_library  | │ │        )                                                                 │ │
godot_private_asset_library  | │ │    f = <built-in function exec>                                          │ │
godot_private_asset_library  | │ │ kwds = {}                                                                │ │
godot_private_asset_library  | │ ╰──────────────────────────────────────────────────────────────────────────╯ │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │ /app/main.py:19 in <module>                                                  │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │    16                                                                        │
godot_private_asset_library  | │    17                                                                        │
godot_private_asset_library  | │    18 project_manager = ProjectManager()                                     │
godot_private_asset_library  | │ ❱  19 asyncio.create_task(project_manager.load_projects())                   │
godot_private_asset_library  | │    20 project_icon_cache = {}                                                │
godot_private_asset_library  | │    21                                                                        │
godot_private_asset_library  | │    22 app = FastAPI()                                                        │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │ ╭───────────────────────────────── locals ─────────────────────────────────╮ │
godot_private_asset_library  | │ │         asyncio = <module 'asyncio' from                                 │ │
godot_private_asset_library  | │ │                   '/usr/local/lib/python3.10/asyncio/__init__.py'>       │ │
godot_private_asset_library  | │ │   default_timer = <built-in function perf_counter>                       │ │
godot_private_asset_library  | │ │        markdown = <module 'readme_renderer.markdown' from                │ │
godot_private_asset_library  | │ │                   '/usr/local/lib/python3.10/site-packages/readme_rende… │ │
godot_private_asset_library  | │ │              os = <module 'os' from '/usr/local/lib/python3.10/os.py'>   │ │
godot_private_asset_library  | │ │ project_manager = <inc.ProjectManager.ProjectManager object at           │ │
godot_private_asset_library  | │ │                   0x76a3d1126380>                                        │ │
godot_private_asset_library  | │ │        settings = Settings(                                              │ │
godot_private_asset_library  | │ │                   │   app_version='0.1.0',                               │ │
godot_private_asset_library  | │ │                   │   app_name='Godot Private Asset Library',            │ │
godot_private_asset_library  | │ │                   │   godot_assets_path_local='godot_assets',            │ │
godot_private_asset_library  | │ │                   │   cache_path_local='cache',                          │ │
godot_private_asset_library  | │ │                   │   zip_path_local='cache/zip',                        │ │
godot_private_asset_library  | │ │                   │   protocol='http',                                   │ │
godot_private_asset_library  | │ │                   │   domain='127.0.0.1',                                │ │
godot_private_asset_library  | │ │                   │   port=8080,                                         │ │
godot_private_asset_library  | │ │                   │   url='http://127.0.0.1:8080'                        │ │
godot_private_asset_library  | │ │                   )                                                      │ │
godot_private_asset_library  | │ │            time = <module 'time' (built-in)>                             │ │
godot_private_asset_library  | │ ╰──────────────────────────────────────────────────────────────────────────╯ │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │ /usr/local/lib/python3.10/asyncio/tasks.py:336 in create_task                │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │   333 │                                                                      │
godot_private_asset_library  | │   334 │   Return a Task object.                                              │
godot_private_asset_library  | │   335 │   """                                                                │
godot_private_asset_library  | │ ❱ 336 │   loop = events.get_running_loop()                                   │
godot_private_asset_library  | │   337 │   task = loop.create_task(coro)                                      │
godot_private_asset_library  | │   338 │   _set_task_name(task, name)                                         │
godot_private_asset_library  | │   339 │   return task                                                        │
godot_private_asset_library  | │                                                                              │
godot_private_asset_library  | │ ╭───────────────────────────────── locals ─────────────────────────────────╮ │
godot_private_asset_library  | │ │ coro = <coroutine object ProjectManager.load_projects at 0x76a3d0c5c740> │ │
godot_private_asset_library  | │ │ name = None                                                              │ │
godot_private_asset_library  | │ ╰──────────────────────────────────────────────────────────────────────────╯ │
godot_private_asset_library  | ╰──────────────────────────────────────────────────────────────────────────────╯
godot_private_asset_library  | RuntimeError: no running event loop
godot_private_asset_library  | sys:1: RuntimeWarning: coroutine 'ProjectManager.load_projects' was never awaited
godot_private_asset_library exited with code 1

 'docker compose -f Docker-compose-local.yml build godot_asset_library'
>WARN[0000] /home/dread81/git/Godot-Private-Asset-Library/Docker-compose-local.yml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion 
no such service: godot_asset_library

**Comment by bison--** [comment]
wo no worries and thanks for the detailed report, will look into it!

**Comment by bison--** [comment]
classic "i don't know how this had ever worked" developer moment.  
I fixed the dockerfile, please rebuild the container with `docker compose -f docker-compose-local.yml build godot_asset_library` and try again.
If it works for you I'll push a new container to dockerhub.

**Comment by GreedyDragon81** [comment]
ok so I got it to work on Mac OS docker and linux docker!  build was successful and on Mac I can access the url on 127.0.0.1, however, when editing the config file and changing the Url to what my unbuntu server is, it still builds with 127.0.0.1. am I supposed to change that elsewhere? 

Let me know if you want this as a separate issue.

Thank you fixing so quickly I am one step closer to self hosting my godot development ecosystem!

**Comment by bison--** [comment]
NICE!  
Based on your problems I extended the README, does this help/does it work doing this https://github.com/SpielmannSpiel/Godot-Private-Asset-Library?tab=readme-ov-file#config ?

**Comment by GreedyDragon81** [comment]
Yes! It works. I did not use docker this time. I installed it via python within linux server and got it working. much easier. I could not get the docker version to change Ip and port, but that might be a me problem. Thank you. I will let you know if I find anything else. This is really cool!

**Comment by bison--** [comment]
nice that it runs now 🎉  
you have to do some backflips for the docker-compose todo that.  
that was why I asked, if this is "enough" for you to apply it to the docker-compose system.  
So I probably should add this to the docker-compose manual ^^"  
What you have to do there is "just normal docker compose stuff", but the "just" part is the tricky part, aka you have to know docker compose well.

**Comment by GreedyDragon81** [comment]
Yes, I had a feeling it was a lack of docker compose knowledge on my end. I have been recently getting into it. But if you do add more instruction that would be awesome! I am going to take time to try compose again this week to figure it out. But seriously thank you for the quick changes! I almost gave up trying to find a solution to self host this.

**Comment by bison--** [comment]
you are very welcome 🥃   
heh, if my stuff is actually useful to someone i'll gladly put some work into it, fixing the buggy bits an pieces that were just "non issues" for me, actually making this better.  
And so I found again some bugs and fixed em. You have to rebuild the container of course.  
Please let me know if the docker compose file example is sufficient https://github.com/SpielmannSpiel/Godot-Private-Asset-Library?tab=readme-ov-file#docker-compose .

**Comment by GreedyDragon81** [comment]
Yes, I just rebuilt the container following this and now I am able to connect to my unbuntu docker container. Just for fun I ran it on my unraid server and even got it working there. Thank you again for doing all these fixes! I will add a test asset to it and see if there are any issues and I will let you know if I find any.

**Comment by bison--** [comment]
NICE!  
But please please please keep in mind to never ever let it run unprotected (eg. Traefik + htaccess https://doc.traefik.io/traefik/middlewares/http/basicauth/ ) on the open internet (except for short tests)! https://github.com/SpielmannSpiel/Godot-Private-Asset-Library?tab=readme-ov-file#warning  
Someone can DDOS the server and literally kill the SSD (if any) by just keep generating new ZIP files / refreshing the file-cache
